softicar/platform
# PLAT-849 Fixed Hot-Deployment for AjaxFramework

- removed useless IAjaxFramework interface
- AjaxFramework.initialize no longer requires a full ServletContext
- extracted code to new project `platform-common-servlet`
- added new IWebServiceFactory interface for source code entry points
  - the service classes themselves are no source code entry points anymore
  - this simplifies the life-cycle of services, they are no longer singletons

Test Plan:
- AjaxDomNodeServerTestDiv
  - start AjaxDomNodeServerTestDiv
  - open page, click and observe code works
  - change label of button in Eclipse and save
  - press F5
  - observe label of button is changed
- HotWebServiceTestServer
  - start change code and observe code changes are effective after session reset
- WebServiceTestServer
  - start and observe it is working (no hot code deployment expected) 

This is the new project:
![image](https://user-images.githubusercontent.com/83839745/168144830-8027ebef-721d-4c88-a097-ad121e8b4f0b.png)
